architecture/2023-12-asyncworker-progressInfo.md

```diff
@@ -0,0 +1,130 @@
+# Title
+
+* **Status**: Pending
+* **Author**: Vinaya Damle (@vinayada1)
+
+## Overview
+
+Currently async worker has no knowledge of the operation progress till the operation is complete. If the operation times out, there is no mechanism in the current design that any information about the operation progress so far can be communicated to the user. As a result, the user only sees the error message that the operation timed out.
+
+The proposal here is to add a mechanism so that the running operation can send out progress information to the async worker. As a result, when the deployment operation eventually times out, along with the timed out message, the user will also be able to see the operation progress so far in the error message which could help in troubleshooting the deployment failure.
+
+## Terms and definitions
+
+AsyncWorker - The current architecture which assigns workers to process queued long-running jobs asynchronously.
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6835
+
+### Goals
+- Enable reporting of operation progress in async worker
+
+### Non-Goals
+- Modify reporting of operation progress in CLI
```

Comment:
I think we can say that this is out of scope of this design doc, but we haven't delivered any value to users **until** you can see progress in the CLI or other UX.